### DISCUSSION TITLE
I want to add new uncertainty analysis capability to IDAES

### DISCUSSION THREAD

**Discussion by udiwekar** [IDAES/idaes-pse#435] [General]
I want to add a new uncertainty analysis capability to IDAES. Can I use a dll generated from other language like fortran, C, or vb to do that?

**Comment by jsiirola** [comment]
The short answer is "yes", there are several ways to call external libraries (DLL / SO / DYNLIB) from Python.  However, the details depend heavily on how the library was compiled and how you want to use the library.  A little more context would help us to give you a more detailed answer.  Can you provide a little more information about what you are planning to do?

**Comment by udiwekar** [comment]
Hi John, This is Urmila Diwekar. I have some code in fortran, some in vb6 or vb, and some in C++. Can you tell me how I can use dlls for these elements. For example the VB code can be replaced by python code but the fortran and C++ codes are very large so may not be able to do that so I can use dll for that.
I just downloaded IDAES and have not used it yet but planning to use it. Thanks. Urmila
Urmila Diwekar, M.Tech. Ph.D.Fellow AIChE, Fellow AIMBETel:(630)-886-3047, ***@***.*** 

    On Thursday, July 22, 2021, 01:43:12 PM CDT, John Siirola ***@***.***> wrote:  
 
 


The short answer is "yes", there are several ways to call external libraries (DLL / SO / DYNLIB) from Python. However, the details depend heavily on how the library was compiled and how you want to use the library. A little more context would help us to give you a more detailed answer. Can you provide a little more information about what you are planning to do?
